From 656393e096c3ded8912e86a9f6b1d9d3faa03356 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Thu, 8 Mar 2018 14:46:07 +0200
Subject: [PATCH 3464/5242] MLK-17580: ARM64: dts: support DSD512 for ak4497

commit  382f0d5a49763e7453a486826157740a2034ef66 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add a dedicated DSD512 pinmux group for DSD512 in order
to eliminate the noise caused by a hight MCLK rate.
With the new option the SAI1 BCLK is routed to codec MCLK pin.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mq-evk-ak4497.dts   |   19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-ak4497.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-ak4497.dts
index 390eacc..d5b28bc 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-ak4497.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-ak4497.dts
@@ -61,13 +61,30 @@
 			>;
 		};
 
+		pinctrl_sai1_dsd512: sai1grp_dsd512 {
+			fsl,pins = <
+				MX8MQ_IOMUXC_SAI1_MCLK_SAI1_TX_BCLK		0xd6
+				MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
+				MX8MQ_IOMUXC_SAI1_RXD7_SAI1_TX_DATA4		0xd6
+				MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7		0xd6
+			>;
+		};
 	};
 };
 
 &sai1 {
-	pinctrl-names = "default", "dsd";
+	pinctrl-names = "default", "dsd", "dsd512";
 	pinctrl-0 = <&pinctrl_sai1_pcm>;
 	pinctrl-1 = <&pinctrl_sai1_dsd>;
+	pinctrl-2 = <&pinctrl_sai1_dsd512>;
 	assigned-clocks = <&clk IMX8MQ_CLK_SAI1_SRC>,
 			<&clk IMX8MQ_CLK_SAI1_DIV>;
 	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL2_OUT>;
-- 
1.7.9.5

